pkg/cluster/sshkeys.go

```diff
@@ -11,33 +11,24 @@ import (
 	"math/big"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database"
 )
 
-func (m *manager) generateSSHKey(ctx context.Context) error {
-	var err error
-	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-		if doc.OpenShiftCluster.Properties.SSHKey == nil {
-			sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
-			if err != nil {
-				return err
-			}
+func (m *manager) ensureSSHKey(ctx context.Context) error {
+	sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
```

Comment:
I think a lot of our other `ensure` or `generate` functions check at the top if the property is nil or empty before performing any work.  Can we do that here as well?

```
if doc.OpenShiftCluster.Properties.SSHKey == nil {
    return nil
}

sshKey, err := rsa.GenerateKey(rand.Reader, 2048)
if err != nil {
    return err
}

var f database.OpenShiftClusterDocumentMutator = func(doc *api.OpenShiftClusterDocument error {
    doc.OpenShiftCluster.Properties.SSHKey = x509.MarshalPKCS1PrivateKey(sshKey)
    return nil
}

m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, f)
return err

Reply:
Tks. In this case, we just want to proceed if SSHKey is nil, so your if condition should be reversed. I now have changed that to make it explicit with an early return. See the latest changes.

Reply:
Yeah my bad typo'd it :)